File: problem_1969A/p1969A.cpp
#include <iostream>
#include <vector>

using namespace std;

int noofinvite(vector<int> &arr)
{
    for (int i = 1; i <= arr.size(); i++)
    {
        if (i == arr[arr[i - 1] - 1])
        {
            return 2;
        }
    }
    return 3;
}

int main()
{
    int nooftestcase;
    cin >> nooftestcase;

    for (int i = 0; i < nooftestcase; i++)
    {
        int noOfFriends;
        cin >> noOfFriends;

        vector<int> arr(noOfFriends);
        for (int j = 0; j < noOfFriends; j++)
        {
            cin >> arr[j];
        }

        cout << noofinvite(arr) << endl;
    }
    return 0;
}

File: problem_1969A/p1969A.java
import java.util.*;

public class p1969A
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int nooftestcase =sc.nextInt();
        int i,j;
        for(i =0;i<nooftestcase;i++)
        {
            int noOfFriends=sc.nextInt();
            int arr[] = new int[noOfFriends];

            for(j=0;j<noOfFriends;j++)
            {
                arr[j]=sc.nextInt();
            }
            System.out.println(noofinvite(arr));
        }
    }
    public static int noofinvite(int[] arr)
    {
        int i;
        for(i=1;i<=arr.length;i++)
        {
            if(i==arr[arr[i-1]-1])
            return 2;
        }
        return 3;
    }
}